### ISSUE TITLE
[BUG] RuntimeError: index 1000000000 is out of bounds

### ISSUE THREAD

**Issue by xonfour** [turboderp/exllamav2#680]
### OS

Linux

### GPU Library

CUDA 12.x

### Python version

3.11

### Pytorch version

2.5.1

### Describe the bug

After the interference has been working cleanly and without problems for a while, this suddenly appears:

```
[...] generator.iterate()
          ^^^^^^^^^^^^^^^^^^^
  File "exllamav2/exl2_env/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "exllamav2/generator/dynamic.py", line 973, in iterate
    self.iterate_gen(results)
  File "exllamav2/generator/dynamic.py", line 1213, in iterate_gen
    job.receive_logits(job_logits)
  File "exllamav2/generator/dynamic.py", line 1817, in receive_logits
    ExLlamaV2Sampler.sample(
  File "exllamav2/generator/sampler.py", line 434, in sample
    ExLlamaV2Sampler.apply_dry(settings, tokenizer, sequence_ids, logits)
  File "exllamav2/generator/sampler.py", line 272, in apply_dry
    logits.scatter_add_(-1, indices, penalties)
RuntimeError: index 1000000000 is out of bounds for dimension 2 with size 131072
```

131072 is my total context/cache size. What is going wrong here? Thanks a lot! +1: 

### Reproduction steps

not easy, happens after some (heavy) usage.

### Expected behavior

generator.iterate() should operate without error

### Acknowledgements

- [X] I have looked for similar issues before submitting this one.
- [X] I understand that the developers have lives and my issue will be answered when possible.
- [X] I understand the developers of this program are human, and I will ask my questions politely.

**Comment by xonfour** [comment]
Yes it is. Could it be possible that I just keep to many image embeddings at the same time?

**Comment by turboderp** [comment]
Turns out it's just a special case I hadn't considered. When the same image appears multiple times in a context and you're using DRY, the sampler tries to apply a penalty to image tokens, and since they're not represented in the logits you get an out-of-bounds error. Should be fixed in dev with the latest commit.

**Comment by xonfour** [comment]
Fix seems to work, no more errors so far. Thanks a lot!
